mipengine/node/monetdb_interface/monet_db_connection.py

```diff
@@ -73,22 +73,26 @@ def execute_with_result(self, query: str) -> List:
         self._connection.commit()
 
         with self.cursor() as cur:
-            cur.execute(query)
+            cur.executemany(query, parameters) if many else cur.execute(
```

Comment:
I think that `executemany` does not work properly with `SELECT` statements as stated here:
https://stackoverflow.com/questions/14142554/sqlite3-python-executemany-select

So, I don't think `executemany` should be an option on the `execute_and_fetchall` .

Please validate, I haven't tested it.

Even if it works, we have to make sure that the `fetchall` will work properly and will fetch the results of all select statements, which I don't think it does.